src/handlers/issue-deduplication.ts

```diff
@@ -0,0 +1,136 @@
+import { IssueSimilaritySearchResult } from "../adapters/supabase/helpers/issues";
+import { Context } from "../types";
+const MATCH_THRESHOLD = 0.95;
+const WARNING_THRESHOLD = 0.5;
+
+export interface IssueGraphqlResponse {
+  node: {
+    title: string;
+    url: string;
+  };
+}
+
+/**
+ * Check if an issue is similar to any existing issues in the database
+ * @param context
+ * @returns true if the issue is similar to an existing issue, false otherwise
+ */
+export async function issueChecker(context: Context): Promise<boolean> {
+  const {
+    logger,
+    payload,
+    adapters: { supabase },
+  } = context;
+
+  const issue = payload.issue;
+
+  //First Check if an issue with more than MATCH_THRESHOLD similarity exists (Very Similar)
+  const similarIssue = await supabase.issue.findSimilarIssues(issue.body + issue.title, MATCH_THRESHOLD, issue.node_id);
+  if (similarIssue && similarIssue?.length > 0) {
+    logger.info(`Similar issue which matches more than ${MATCH_THRESHOLD} already exists`);
```

Comment:
Needs to return the results of issues

Reply:
@0x4007 This function handles only the deduplication process, if there's a match it closes the issue, if there's a warning condition, it would create the comment. I am not sure why the similarIssue would be needed in other functions like `add-comment`, etc.

Reply:
It needs to always let the user know which it thinks is similar. It can be wrong

Reply:
So, for both `Match` and `Warning` threshold, there has to be comment with similar issues. Is that right ?